src/linux/classes/com/alibaba/wisp/engine/WispEngine.java

```diff
@@ -344,6 +344,25 @@ public void getCpuTime(long[] ids, long[] times) {
                     }
                 }
             }
+
+            @Override
+            public int poll(SelectableChannel channel, int interestOps, long millsTimeOut) throws IOException {
```

Comment:
1. WAE.poll should return polled event count;
2. WAE.poll receive NET.* events represent as parameters will be clearer.
3. Coroutines are not interrupted when the socket closes. Leave a TODO there

Reply:
OK. please have another look.

Close socket is ok here, because we would check isOpen() every time.